### PR TITLE
Adding Complaint Form

### PR DESCRIPTION
**Pull request by PragyaPR06**
Things to do:
1)Style the dropdown arrow element of select tag
2)Add the loader for the image upload
3)Implement Axios

### CODE REVIEW

File: src/App.jsx
Comment (hereisadi) [inline src/App.jsx:1]: The filename "multipleinputs.jsx"should be in "PASCAL_CASE" pattern

File: src/App.jsx [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
 function App() {
+
   return (
     <>
       <BrowserRouter>
         <Navbar />
         <div>
           <Routes>
             <Route exact path="/" element={<Home />} />
+            <Route exact path="/Complaintform" element={<CForm />} />
```
Comment (hereisadi) [inline src/App.jsx:21]: route "/complaintform" should be in lower case

File: src/Components/Shared/Form/multipleinputs.jsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
+
+import { useState } from "react";
+import styles from "./styleform.module.scss";
+
+
+export default function CForm(){
+    const[Formdata,setformdata]=useState({
```
Comment (hereisadi) [inline src/Components/Shared/Form/multipleinputs.jsx:7]: although the way you wrote would work, but the proper syntax of using useState Hook is:
const [formData, setFormData] = useState({})

File: src/Components/Shared/Form/multipleinputs.jsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        ScholarID :"",
+        Hostel :"",
+        RoomNo :"",
+        Category:"",
+        description:"",
+        Forwardedto:"",
+        Imagefile:""
+    })
+
+    const[Storedata,setstoredata]=useState([]);
```
Comment (hereisadi) [inline src/Components/Shared/Form/multipleinputs.jsx:18]: same syntax issue with useState

File: src/Components/Shared/Form/multipleinputs.jsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
+
+import { useState } from "react";
+import styles from "./styleform.module.scss";
+
+
+export default function CForm(){
```
Comment (hereisadi) [inline src/Components/Shared/Form/multipleinputs.jsx:6]: Use arrow function: (everywhere)
const CForm = () => {}
export default CForm;

File: src/Components/Shared/Form/multipleinputs.jsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    })
+
+    const[Storedata,setstoredata]=useState([]);
+
+    function handleInput(e){
+        const name = e.target.name;
+        const value = e.target.value;
+        // console.log(name, value);
+        setformdata({
+            ...Formdata,
```
Comment (hereisadi) [inline src/Components/Shared/Form/multipleinputs.jsx:25]: use functional update form of setFormData

File: src/Components/Shared/Form/multipleinputs.jsx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+                            </div>
+                            <div className={styles.form_group}>
+                                <textarea type="text" id="description" value={Formdata.description} name="description" onChange={handleInput} autoComplete="off" rows="5" cols="40" required/>
+                                <label htmlFor="description">Description</label>
+                            </div>
+                            <div className={styles.photoUpload}>
+                                    <p>Upload Your Photo</p>
+                                    <div className={styles.photoupload_inner} onDragOver={handleDragOver} 
+                                    onDrop={handleDrop} >
+                                            <img src="https://res.cloudinary.com/dlx4meooj/image/upload/c_pad,b_auto:predominant,fl_preserve_transparency/v1694535132/UC%20VYATHA/Frame_58066_1_nnkr62.jpg?_s=public-apps" alt="" draggable="true" onDragStart={(e)=>e.preventDefault()}></img>
```
Comment (hereisadi) [inline src/Components/Shared/Form/multipleinputs.jsx:139]: img is self closing tag. no need to add explicit closing tag

File: src/Components/Shared/Form/multipleinputs.jsx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+                                        <div className={styles.photouploadcontent}>
+                                            <span className={styles.Dragdrop}>Drag and Drop File</span>
+                                            <span className={styles.or}>-OR-</span>
+                                        </div>
+                                        
+                                        <label id="Browsebutton">
+                                            BROWSE
+                                            <input type="file" name="Imagefile" id="imagebrowse" accept=".jpg, .jpeg, .png" onChange={handleFileInputChange}  required />
```
Comment (hereisadi) [inline src/Components/Shared/Form/multipleinputs.jsx:147]: accept webp file too

File: src/Components/Shared/Form/multipleinputs.jsx [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+        e.preventDefault();
+    }
+
+    function handleDrop(e){
+        // console.log('drop');
+        e.preventDefault();
+
+        const file = e.dataTransfer.files[0];
+
+        if(file){
```
Comment (hereisadi) [inline src/Components/Shared/Form/multipleinputs.jsx:48]: use try catch block for exceptional error handling and if file size is greater than say 250KB throw an alert with proper error msg

File: src/App.jsx
Comment (hereisadi) [inline src/App.jsx:1]: image
select parent's div width is overflowing

File: src/Components/Shared/Form/styleform.module.scss [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+    display: flex;
+    gap:0.5vh;
+    justify-content: center;
+    align-items: center;
+    height: 100%;
+        .Title{position: relative;
+            display:flex;
+            align-items: center;
+            justify-content: center;
+            background: linear-gradient(270deg, #40BDB6 2.67%, #347EC5 119.87%);
```
Comment (hereisadi) [inline src/Components/Shared/Form/styleform.module.scss:15]: declare the gradient in index.scss and use it using var()

File: src/Components/Shared/Form/styleform.module.scss [hunk 10]
```diff
@@ -0,0 +1,185 @@
+body {
```
Comment (hereisadi) [inline src/Components/Shared/Form/styleform.module.scss:1]: don't apply styling to body directly.

File: src/Components/Shared/Form/styleform.module.scss [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+    font-family: Arial, Helvetica, sans-serif;
+    background-color: #f4f4f4;
+    margin: 0;
+    padding: 0;
```
Comment (hereisadi) [inline src/Components/Shared/Form/styleform.module.scss:5]: redundant code. margin and padding has already been set to zero in index.scss

File: src/Components/Shared/Form/styleform.module.scss [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+            font-size: 4vw;
+            height:6vh;
+
+        }
+    
+        .CForm {
+            background-color: #fff;
+            border-radius: 5px;
+            box-shadow: 0 2px 5px rgba(0, 0, 0, 0.3);
+            padding: 20px;
```
Comment (hereisadi) [inline src/Components/Shared/Form/styleform.module.scss:27]: use relative font units

**Review by hereisadi — CHANGES_REQUESTED** [review]

File: src/Components/Shared/Form/MultipleInputs.jsx [hunk 13]
```diff
@@ -0,0 +1,268 @@
+/* eslint-disable */
```
Comment (hereisadi) [inline src/Components/Shared/Form/MultipleInputs.jsx:1]: no need to disable the eslint

**Comment by hereisadi** [comment]
remaining conversation changes has not been fixed yet. @PragyaPR06

**Review by hereisadi — CHANGES_REQUESTED** [review]

File: src/Components/Shared/Form/MultipleInputs.jsx [hunk 14]
```diff
@@ -0,0 +1,268 @@
...
+              <option value="BH9C" id="BH9C" name="Hostel" onChange={handleInput}>
+                BH9C
+              </option>
+              <option value="BH9D" id="BH9D" name="Hostel" onChange={handleInput}>
+                BH9D
+              </option>
+              <option value="PGH" id="PGH" name="Hostel" onChange={handleInput}>
+                PGH
+              </option>
+              <option value="PGH" id="PGH" name="Hostel" onChange={handleInput}>
```
Comment (Himangshu-101) [inline src/Components/Shared/Form/MultipleInputs.jsx:156]: why did you give PGH in name and value for gh1, gh2, gh3 ?
wouldn't it be different for each ?

**Review by Himangshu-101 — DISMISSED** [review]

**Review by hereisadi — APPROVED** [review]

**Pull request merged**